src/api/app/controllers/person/token_controller.rb

```diff
@@ -7,25 +7,34 @@ class TokenController < ApplicationController
     def index
       authorize @user, :show?
 
-      @list = @user.service_tokens
+      @list = @user.tokens
     end
 
     # POST /person/<login>/token
     def create
       authorize @user, :update?
 
-      pkg = nil
-      if params[:project] || params[:package]
-        pkg = Package.get_by_project_and_name(params[:project], params[:package])
-      end
-      @token = Token::Service.create(user: @user, package: pkg)
+      pkg = if params[:project] || params[:package]
+              Package.get_by_project_and_name(params[:project], params[:package])
+            end
+
+      token_type = case params[:operation]
+                   when 'rebuild'
+                     Token::Rebuild
+                   when 'release'
+                     Token::Release
+                   else
+                     # default is Token::Service
+                     Token::Service
+                   end
```

Comment:
This looks like it could be dried with the code [here](https://github.com/openSUSE/open-build-service/pull/6965/files#diff-00232d8e1ee1f7438e6cbb963c74a038R83)

Reply:
yup. Did it now.